mypy/find_sources.py

```diff
@@ -15,19 +16,20 @@ class InvalidSourceList(Exception):
     """Exception indicating a problem in the list of sources given to mypy."""
 
 
-# TODO: use a FileSystemCache for this
-def create_source_list(files: Sequence[str], options: Options) -> List[BuildSource]:
+def create_source_list(files: Sequence[str], options: Options,
+                       fscache: Optional[FileSystemMetaCache]=None) -> List[BuildSource]:
     """From a list of source files/directories, makes a list of BuildSources.
 
     Raises InvalidSourceList on errors.
     """
+    fscache = fscache or FileSystemMetaCache()
     targets = []
     for f in files:
         if f.endswith(PY_EXTENSIONS):
             # Can raise InvalidSourceList if a directory doesn't have a valid module name.
-            targets.append(BuildSource(f, crawl_up(f)[1], None))
-        elif os.path.isdir(f):
-            sub_targets = expand_dir(f)
+            targets.append(BuildSource(f, crawl_up(fscache, f)[1], None))
```

Comment:
For clarity, change `import os` above to `import os.path` since we no longer directly access any functions in `os` in this module.